third_party/terraform/utils/iam_bigquery_dataset.go

```diff
@@ -0,0 +1,233 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+var IamBigqueryDatasetSchema = map[string]*schema.Schema{
+	"dataset_id": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"project": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+		ForceNew: true,
+	},
+}
+
+var bigqueryAccessPrimitiveToRoleMap = map[string]string{
```

Comment:
Interesting - maybe we should move the opposite RoleToPrimativeMap here as well

Reply:
The opposite map has the DSF that uses it alongside it, so I'd prefer to keep that together